app/data_client.go

```diff
@@ -0,0 +1,763 @@
+	RobotName        string
+	RobotID          string
+	PartName         string
+	PartID           string
+	ComponentType    string
+	ComponentName    string
+	MethodName       string
+	MethodParameters map[string]interface{}
+	Tags             []string
+	MimeType         string
+}
+
+// CaptureInterval describes the start and end time of the capture in this file.
+type CaptureInterval struct {
+	Start time.Time
+	End   time.Time
+}
+
+// TabularData contains data and metadata associated with tabular data.
+type TabularData struct {
+	Data          map[string]interface{}
+	MetadataIndex uint32
+	Metadata      CaptureMetadata
+	TimeRequested time.Time
+	TimeReceived  time.Time
+}
+
+// BinaryData contains data and metadata associated with binary data.
+type BinaryData struct {
+	Binary   []byte
+	Metadata BinaryMetadata
+}
+
+// BinaryMetadata is the metadata associated with binary data.
+type BinaryMetadata struct {
+	ID              string
+	CaptureMetadata CaptureMetadata
+	TimeRequested   time.Time
+	TimeReceived    time.Time
+	FileName        string
+	FileExt         string
+	URI             string
+	Annotations     Annotations
+	DatasetIDs      []string
+}
+
+// BinaryID is the unique identifier for a file that one can request to be retrieved or modified.
+type BinaryID struct {
+	FileID         string
+	OrganizationID string
+	LocationID     string
+}
+
+// BoundingBox represents a labeled bounding box on an image.
+// x and y values are normalized ratios between 0 and 1.
+type BoundingBox struct {
+	ID             string
+	Label          string
+	XMinNormalized float64
+	YMinNormalized float64
+	XMaxNormalized float64
+	YMaxNormalized float64
+}
+
+// Annotations are data annotations used for machine learning.
+type Annotations struct {
+	Bboxes []BoundingBox
+}
+
+// TabularDataReturn represents the result of a TabularDataByFilter query.
+// It contains the retrieved tabular data and associated metadata,
+// the total number of entries retrieved (Count), and the ID of the last returned page (Last).
+type TabularDataReturn struct {
+	TabularData []TabularData
+	Count       uint64
+	Last        string
+}
+
+// BinaryDataReturn represents the result of a BinaryDataByFilter query.
+// It contains the retrieved binary data and associated metadata,
+// the total number of entries retrieved (Count), and the ID of the last returned page (Last).
+type BinaryDataReturn struct {
+	BinaryData []BinaryData
+	Count      uint64
+	Last       string
+}
+
+// DatabaseConnReturn represents the response returned by GetDatabaseConnection.
+// It contains the hostname endpoint, a URI for connecting to the MongoDB Atlas Data Federation instance,
+// and a flag indicating whether a database user is configured for the Viam organization.
+type DatabaseConnReturn struct {
+	Hostname        string
+	MongodbURI      string
+	HasDatabaseUser bool
+}
+
+// NewDataClient constructs a new DataClient using the connection passed in by the viamClient and the provided logger.
```

Comment:
Just wanted to note that it says that there's a logger, but there is no logger mentioned in the function.